pkgs/development/compilers/graalvm/community-edition/default.nix

```diff
@@ -1,17 +1,60 @@
-{ lib, callPackage, Foundation }:
+{ callPackage, Foundation }:
+/*
+  Add new graal versions and products here and then see update.nix on how to
+  generate the sources.
+*/
 
 let
   mkGraal = opts: callPackage (import ./mkGraal.nix opts) {
     inherit Foundation;
   };
+
+  /*
+    Looks a bit ugly but makes version update in the update script using sed
+    much easier
+  */
+  graalvm11-ce-release-version = "22.0.0.2";
+  graalvm17-ce-release-version = "22.0.0.2";
+  graalvm11-ce-dev-version = "22.2.0-dev-20220401_1942";
+  graalvm17-ce-dev-version = "22.2.0-dev-20220401_1942";
```

Comment:
Minor nitpick but I think we can just pass this values directly to the `mkGraal` function now. There is no more unnecessary repetition and I think it makes easier to understand which version is being used where.

Reply:
Yes, but then the version replacement in the update script gets much more complex:

E.g I then need to use some awk or sed magic to find the graalvm11-ce line and replace the defaultVersion or version string exactly once and the same if the update script is invoked with graalvm17.

Alternatively, I could probably also do that version update using nix. I need to think about this.

Reply:
Ah ok, this seems fair.